### PR TITLE
ISPN-15016 ZMPOP

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-15016

### CODE REVIEW

File: server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/ZMPOP.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+         }
+
+         if (!iteNames.hasNext()) {
+            return handler.stageToReturn(CompletableFutures.completedNull(), ctx, Consumers.GET_BICONSUMER);
+         }
+
+         byte[] nextName = iteNames.next();
+         return asyncCalls(handler.getSortedSeMultimap().pop(nextName, isMin, count), nextName, iteNames, count, isMin, ctx, handler);
+      });
+   }
```
[the remark this answers is not in the document]
Comment (karesti) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/ZMPOP.java:140]: we need to change this in some list commands too then

**Comment by karesti** [comment]
@jabolina I added your commit

File: server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/ZMPOP.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+         }
+      }
+
+      if (arguments.size() > pos) {
+         RespErrorUtil.syntaxError(handler.allocator());
+         return handler.myStage();
+      }
+
+      CompletionStage<List<Object>> cs = asyncCalls(CompletableFutures.completedNull(), null, sortedSetNames.iterator(), count, isMin, ctx, handler);
+      return handler.stageToReturn(cs, ctx, RESPONSE_HANDLER);
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/ZMPOP.java:111]: This still needs to declare `REPONSE_HANDLER` as:

```java
   private static final BiConsumer<List<Object>, ByteBufPool> RESPONSE_HANDLER = (res, buff) -> {
      if (res != null) Consumers.ZMPOP_BICONSUMER.accept(res, buff);
      else Consumers.GET_BICONSUMER.accept(null, buff);
   };
```

**Review by jabolina — APPROVED** [review]
I see some of the test failures were already fixed on main. Also, the ZSCAN should be fixed once integrated with the main changes we added recently on #11502.

**Pull request merged**